File: Courses/PHP Web Basics - май 2019/18. Exercise Polymorphism, Interfaces and Abstraction/01.07.FoodShortage/Citizen.php
<?php

include_once "IdentifiableInterface.php";
include_once "BirthdayInterface.php";
include_once "BuyerInterface.php";

class Citizen implements IdentifiableInterface, BirthdayInterface, BuyerInterface
{

    private $name;
    private $age;
    private $id;
    private $birthday;
    private $food = 0;


    public function __construct(string $name, int $age, string $id, string $birthday)
    {
        $this->name = $name;
        $this->age = $age;
        $this->id = $id;
        $this->birthday = $birthday;
    }


//    public function getName(): string
//    {
//        return $this->name;
//    }

    public function getId(): string
    {
        return $this->id;
    }

    public function getBirthday(): string
    {
        return $this->birthday;
    }

    public function BuyFood()
    {
        $this->setFood($this->getFood() + 10);
    }
    public function getFood(): int
    {
        return $this->food;
    }
    public function setFood(int $value)
    {
        $this->food = $value;
    }
}

File: Courses/C# OOP/Basics/12. Interfaces and Abstraction Exercise/Interfaces and Abstraction - Exercise/06.BirthdayCelebrations/Citizen.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace _06.BirthdayCelebrations
{
    class Citizen : IIdentifiable, IBirthable
    {
        public Citizen(string name, int age, string id, string birthday)
        {
            Name = name;
            Age = age;
            Id = id;
            Birthday = birthday;
        }

        public string Name { get; set; }
        public int Age { get; set; }
        public string Id { get; set; }
        public string Birthday { get; set; }
       
    }
}
